samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/AdapterWithErrorHandler.java

```diff
@@ -0,0 +1,104 @@
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.corebot.app.insights;
+
+import com.microsoft.bot.applicationinsights.core.TelemetryInitializerMiddleware;
+import com.microsoft.bot.builder.BotTelemetryClient;
+import com.microsoft.bot.builder.ConversationState;
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.connector.Channels;
+import com.microsoft.bot.integration.BotFrameworkHttpAdapter;
+import com.microsoft.bot.integration.Configuration;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.ActivityTypes;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An Adapter that provides exception handling.
+ */
+public class AdapterWithErrorHandler extends BotFrameworkHttpAdapter {
+
+    private static final String ERROR_MSG_ONE = "The bot encountered an error or bug.";
+    private static final String ERROR_MSG_TWO =
+        "To continue to run this bot, please fix the bot source code.";
+    // Create field for telemetry client. Add IBotTelemetryClient parameter to AdapterWithErrorHandler
+    private BotTelemetryClient adapterBotTelemetryClient;
+
+    /**
+     * Constructs an error handling BotFrameworkHttpAdapter by providing an
+     * {@link com.microsoft.bot.builder.OnTurnErrorHandler}.
+     *
+     * <p>
+     * For this sample, a simple message is displayed. For a production Bot, a more
+     * informative message or action is likely preferred.
+     * </p>
+     *
+     * @param withConfiguration The Configuration object to use.
+     */
+    public AdapterWithErrorHandler(
+        Configuration withConfiguration,
+        TelemetryInitializerMiddleware telemetryInitializerMiddleware,
+        BotTelemetryClient botTelemetryClient,
+        @Nullable ConversationState withConversationState) {
+        super(withConfiguration);
+        this.use(telemetryInitializerMiddleware);
+
+        // Use telemetry client so that we can trace exceptions into Application Insights
+        this.adapterBotTelemetryClient = botTelemetryClient;
+        setOnTurnError((turnContext, exception) -> {
+            // Track exceptions into Application Insights
+            // Set up some properties for our exception tracing to give more information
+            Map<String, String> properties = new HashMap<String, String>();
+            properties.put("Bot exception caught in", "AdapterWithErrorHandler - OnTurnError");
+
+            // Send the exception telemetry:
+            adapterBotTelemetryClient.trackException((Exception) exception, properties, null);
+            LoggerFactory.getLogger(AdapterWithErrorHandler.class).error("onTurnError", exception);
```

Comment:
Here is missing these [comments](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/21.corebot-app-insights/AdapterWithErrorHandler.cs#L39).